### PR TITLE
Make notebook props explicit

### PR DESCRIPTION
**Pull request by oisincoveney**
Makes props for `Notebook` components explicit instead of prop-spread-drilling.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/notebook/Notebook.tsx [hunk 1]
```diff
@@ -86,27 +84,5 @@ const Notebook = ({ className, updateQuestion, ...props }: NotebookProps) => {
...
-
-  return undefined;
-}
-
 // eslint-disable-next-line import/no-default-export -- deprecated usage
-export default _.compose(
-  Questions.load({
-    id: (state: State, { question }: NotebookProps) =>
-      getSourceQuestionId(question),
-    entityAlias: "sourceQuestion",
```
Comment (oisincoveney) [inline frontend/src/metabase/query_builder/components/notebook/Notebook.tsx:109]: The alias here specifies that `sourceQuestion` is passed into the component, but we don't use it in the components so there's no point in having this extra call.

File: frontend/src/metabase/query_builder/components/notebook/Notebook.tsx [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-import {
-  getQuestionIdFromVirtualTableId,
-  isVirtualCardId,
-} from "metabase-lib/v1/metadata/utils/saved-questions";
-import type { State } from "metabase-types/store";
 
 import { NotebookSteps } from "./NotebookSteps";
 
-interface NotebookProps {
-  className?: string;
```
Comment (oisincoveney) [inline frontend/src/metabase/query_builder/components/notebook/Notebook.tsx:19]: this never gets used

File: frontend/src/metabase/query_builder/components/notebook/Notebook.tsx [hunk 3]
```diff
@@ -76,7 +69,12 @@ const Notebook = ({ className, updateQuestion, ...props }: NotebookProps) => {
 
   return (
     <Box pos="relative" p={{ base: "1rem", sm: "2rem" }}>
-      <NotebookSteps updateQuestion={handleUpdateQuestion} {...props} />
+      <NotebookSteps
```
Comment (oisincoveney) [inline frontend/src/metabase/query_builder/components/notebook/Notebook.tsx:72]: `NotebookSteps` explicitly specifies its props so this is a safe change

File: frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx [hunk 4]
```diff
@@ -18,13 +20,22 @@ import { Flex, Box, rem } from "metabase/ui";
...
 
-interface NotebookContainerProps {
+type NotebookContainerProps = {
   isOpen: boolean;
-}
+} & NotebookProps;
 
 export const NotebookContainer = ({
   isOpen,
-  ...props
```
Comment (oisincoveney) [inline frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx:27]: `props` was being spread to `Notebook` so the only props that could be in here are `NotebookProps`

File: frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx [hunk 5]
```diff
@@ -65,7 +76,9 @@ export const NotebookContainer = ({
 
   const Handle = forwardRef<
     HTMLDivElement,
-    Partial<ResizableBoxProps> & { onResize?: any } //Mantine and react-resizeable have different opinions on what onResize should be
+    Partial<ResizableBoxProps> & {
+      onResize?: any;
+    } //Mantine and react-resizeable have different opinions on what onResize should be
```
Comment (npretto) [inline frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx:81]:
```suggestion
      onResize?: any; //Mantine and react-resizeable have different opinions on what onResize should be
    }
```

**Review by npretto — APPROVED** [review]
Left a small comment but otherwise looks good

**Comment by npretto** [comment]
@oisincoveney I think this should be backported though?

**Pull request merged**